### PR TITLE
Switch to CircleCI v2

### PR DESCRIPTION
**Pull request by davidfischer**
- v1 is sunsetted completely and no longer building whatsoever

### CODE REVIEW

**Comment by davidfischer** [comment]
This is interesting. For some reason, CircleCI can't clone the repository from github. I've never seen this with a public repository!

**Comment by davidfischer** [comment]
After resetting the project in CircleCI, it re-added the necessary webhook and deploy key and ran successfully.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            pip install coverage codecov
+
+      - save_cache:
+          paths:
+            - ./.venv
+          key: v1-dependencies-{{ checksum "requirements/local.txt" }}
+
+      # run tests!
+      - run:
+          name: run tests
```
Comment (mmaybeno) [inline .circleci/config.yml:38]: I needed to add these env variables to get it to pass.
```
environment:
  CIRCLE_PROJECT_USERNAME: pythonsd
  CIRCLE_PROJECT_REPONAME: pythonsd-django
```

**Review by mmaybeno — COMMENTED** [review]
After adding env variables everything passed on my end. A few warnings about cache errors: 
```
====>> Saving Cache
Error: 
Skipping cache - error checking storage: not supported
```
But overall looked good.

Ran these commands after adding circleci cli:
1. `circleci config validate`
2. `circleci local execute --job build`

**Comment by davidfischer** [comment]
I wonder if these warnings are related to the local `circleci` command line client. I don't see any warnings in the [actual CircleCI output](https://circleci.com/gh/pythonsd/pythonsd-django/56).

**Comment by mmaybeno** [comment]
Ya maybe those are globally set on the cloud runner.

**Comment by mmaybeno** [comment]
Actually yes, in the `Spin up environment` step you can see those set. I'm fine with leaving them out then.

**Review by mmaybeno — APPROVED** [review]

**Pull request merged**